iguana.commons/src/main/java/org/aksw/iguana/commons/io/BigByteArrayInputStream.java

```diff
@@ -0,0 +1,54 @@
+package org.aksw.iguana.commons.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class BigByteArrayInputStream extends InputStream {
+
+    private BigByteArrayOutputStream bbaos;
+
+    private byte[] curArray;
+    private int curPos=0;
+    private int curPosInArray=0;
+
+    public BigByteArrayInputStream(byte[] bytes) throws IOException {
+        bbaos = new BigByteArrayOutputStream();
+        bbaos.write(bytes);
+        setNextArray();
+    }
+
+    public BigByteArrayInputStream(BigByteArrayOutputStream bbaos){
+        this.bbaos = bbaos;
+        setNextArray();
+    }
+
+    private void setNextArray(){
+        curArray=bbaos.getBaos().get(curPos++).toByteArray();
+    }
+
+    @Override
+    public int read() throws IOException {
+        if(eos()){
+            return -1;
+        }
+        int ret;
+        //Is it Integer.MAX_VALUE or Integer.MAX_VALUE-1?
```

Comment:
@bigerl 
I cannot wrap my head around it atm. Can you confirm it?

Reply:
must be `Integer.MAX_VALUE - 3`